src/main/java/org/rcsb/cif/model/ModelFactory.java

```diff
@@ -207,14 +330,9 @@ public static Column createColumnText(String categoryName,
     @SuppressWarnings("unchecked")
     public static Column createColumnBinary(String categoryName, String columnName, Map<String, Object> encodedColumn) {
         Object binaryData = Codec.decode((Map<String, Object>) encodedColumn.get("data"));
-        boolean isIntArray = binaryData instanceof int[];
-        boolean isDoubleArray = binaryData instanceof double[];
-        int rowCount = isIntArray ? ((int[]) binaryData).length :
-                isDoubleArray ? ((double[]) binaryData).length : ((String[]) binaryData).length;
-        boolean hasMask = encodedColumn.containsKey("mask") && encodedColumn.get("mask") != null &&
-                !((Map) encodedColumn.get("mask")).isEmpty();
-        int[] mask = hasMask ? (int[]) Codec.decode((Map<String, Object>) encodedColumn.get("mask")) : null;
-
+        int rowCount = Array.getLength(binaryData);
```

Comment:
Oh, nice. Didn't know that there is an easy way to get the length of both primitive and Object arrays when their type isn't specified.